website/content/en/docs/kubebuilder/references/client.md

```diff
@@ -0,0 +1,574 @@
+
+Example:
+
+```Go
+import (
+	"context"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	pod := &v1.Pod{}
+	err := r.Get(context.TODO(), request.NamespacedName, pod)
+
+	...
+
+	ctx := context.TODO()
+	if pod.Status.Phase == v1.PodUnknown {
+		// Delete the pod after 5 seconds.
+		err := r.Delete(ctx, pod, client.GracePeriodSeconds(5))
+		...
+	}
+
+	...
+}
+```
+
+[delete-opts]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#DeleteOptions
+[grace-period-seconds]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#GracePeriodSeconds
+[preconditions]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#Preconditions
+[propagation-policy]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#PropagationPolicy
+
+#### DeleteAllOf
+
+```Go
+// DeleteAllOf deletes all objects of the given type matching the given options.
+func (c Client) DeleteAllOf(ctx context.Context, obj runtime.Object, opts ...client.DeleteAllOfOption) error
+```
+
+A `client.DeleteAllOfOption` is an interface that sets [`client.DeleteAllOfOptions`][deleteallof-opts] fields. A `client.DeleteAllOfOption` wraps a [`client.ListOption`](#list) and [`client.DeleteOption`](#delete).
+
+Example:
+
+```Go
+import (
+	"context"
+	"fmt"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	// Delete all pods in the request namespace with a label of `app=<name>`
+	// and phase `Failed`.
+	pod := &v1.Pod{}
+	opts := []client.DeleteAllOfOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"app", request.NamespacedName.Name},
+		client.MatchingFields{"status.phase": "Failed"},
+		client.GracePeriodSeconds(5),
+	}
+	ctx := context.TODO()
+	err := r.DeleteAllOf(ctx, pod, opts...)
+
+	...
+}
+```
+
+[deleteallof-opts]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#DeleteAllOfOptions
+
+### Example usage
+
+```Go
+import (
+	"context"
+	"reflect"
+
+	appv1alpha1 "github.com/example-org/app-operator/pkg/apis/app/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type ReconcileApp struct {
+    client.Client
+    Log    logr.Logger
+    Scheme *runtime.Scheme
+}
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Update for new scaffold. `request reconcile.Request`==>`req ctrl.Request`
Update the imports as well `"sigs.k8s.io/controller-runtime/pkg/reconcile"`==>`ctrl "sigs.k8s.io/controller-runtime"`